src/Microsoft.DotNet.Tools.Compiler.Csc/Program.cs

```diff
@@ -58,7 +68,14 @@ public static int Main(string[] args)
             }
             catch (ArgumentSyntaxException)
             {
-                return ExitFailed;
+                help = true;
+            }
+
+            if (help)
+            {
+                Console.WriteLine(helpText);
+
+                return 0;
```

Comment:
Are there cases where we might not want to return 0? For example if someone has typed invalid input (ArgumentException got thrown) wouldn't we want to return non-zero exit code?